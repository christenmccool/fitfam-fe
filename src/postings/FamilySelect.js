import React, {useState} from 'react';

import Box from '@mui/material/Box';
import Grid from '@mui/material/Grid';
import TextField from '@mui/material/TextField';
import MenuItem from '@mui/material/MenuItem';

/** Form to update user profile
 *  
 * Routed at /profile/update
 */
const FamilySelect = ({ families, familyId, changeFamilyId }) => {
  const [famId, setFamId] = useState(familyId)

  const familyOptions = families.map(ele => (
     {
       value: ele.familyId,
       label: ele.familyName
      }
    )
  )

  const handleChange =  async (event) => {
    const {value} = event.target;
    setFamId(value);
    changeFamilyId(value);
  }

  return (
      <Box>
        <TextField
          id="primFamId"
          select
          label="Select FitFam"
          value={famId}
          onChange={handleChange}
          sx={{backgroundColor: "#FFF", minWidth: 260}}
          InputProps={{style: {fontSize: '20px'}}}
        >
          {familyOptions.map((option) => (
            <MenuItem key={option.value} value={option.value}> 
              {option.label}
            </MenuItem>
          ))}
        </TextField>
    </Box>
  )
}

export default FamilySelect;