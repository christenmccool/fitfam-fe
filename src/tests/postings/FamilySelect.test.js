import React from "react";
import { render, screen } from "@testing-library/react";
import UserEvent from "@testing-library/user-event";
import { MemoryRouter } from "react-router-dom";

import FamilySelect from "../../postings/FamilySelect";

const families = [
  {
    familyId: 1,
    familyName: "Family 1"
  },
  {
    familyId: 2,
    familyName: "Family 2"
  }
];

it('renders without crashing', function () {
  render(
    <MemoryRouter>
      <FamilySelect families={families} familyId={1} />
    </MemoryRouter>
  );
});


it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <FamilySelect families={families} familyId={1} />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});


it("renders family select", function () {
  render(
    <MemoryRouter>
      <FamilySelect families={families} familyId={1} />
    </MemoryRouter>
  );
  expect(screen.queryByLabelText("Select FitFam")).toBeInTheDocument();  
});


it("responds to user input", function () {
  render(
    <MemoryRouter>
      <FamilySelect families={families} familyId={1} />
    </MemoryRouter>
  );

  const categorySelect = screen.getByRole('button', {name: /select fitfam/i});
  UserEvent.click(categorySelect);
  UserEvent.click(screen.getByText(/family 2/i));
  expect(categorySelect).toHaveTextContent(/family 2/i);
});

