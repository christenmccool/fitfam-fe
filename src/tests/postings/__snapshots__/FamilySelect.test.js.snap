// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-0"
  >
    <div
      class="MuiFormControl-root MuiTextField-root css-fuzc6w-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
        for="primFamId"
        id="primFamId-label"
      >
        Select FitFam
      </label>
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        style="font-size: 20px;"
        variant="outlined"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-labelledby="primFamId-label primFamId"
          class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
          id="primFamId"
          role="button"
          tabindex="0"
        >
          Family 1
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
          tabindex="-1"
          value="1"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-7nxzgd"
          >
            <span>
              Select FitFam
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
